devstack/devstack.dockerfile

```diff
@@ -0,0 +1,80 @@
+FROM ubuntu:xenial
+ARG USER_NAME
+ENV USER_NAME ${USER_NAME:-root}
+ARG USER_ID
+ENV USER_ID ${USER_ID:-0}
+
+#### Docker specific ###########################################################
+#
+# en_US.UTF-8 is assumed to always be installed
+#
+RUN apt-get update && apt-get install -y locales && locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8
+#
+# Install dependencies
+#
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y sudo ca-certificates openssh-server
+#
+# All commands are run via exec under a non privileged user that has the same name as the user building the image
+# and the same id. So that volumes mounted from a directory they own are not unexpectedly chowned to root in various
+# places.
+#
+RUN if test $USER_NAME != root ; then useradd --no-create-home --home-dir /tmp --uid $USER_ID $USER_NAME && echo "$USER_NAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers ; fi
+#
+#### Install dependencies ######################################################
+#
+# Firefox 52 for selenium compatibility
+#
+RUN apt-get install -y software-properties-common && add-apt-repository --yes ppa:jonathonf/firefox-esr-52 && apt-get update && apt-get install -y firefox-esr
+#
+# Install postgresql 10 because ansible-playbooks/playbooks/group_vars/postgres/public.yml
+# has version 10 and ansible-playbooks/playbooks/roles/postgres/tasks/main.yml installs it
+# 
+RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ xenial-pgdg main' >> /etc/apt/sources.list.d/pgdg.list && wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y postgresql-10
+#
+# Install consul
+#
+RUN apt-get install -y unzip && wget -P /tmp https://releases.hashicorp.com/consul/1.2.1/consul_1.2.1_linux_amd64.zip && unzip /tmp/consul_1.2.1_linux_amd64.zip -d /usr/local/bin
+COPY devstack/consul.service /etc/systemd/system/consul.service
```

Comment:
The approach here should be to start Consul as a separate container within the same network as Ocim through docker-compose, which manages this easily. Same with postgres and basically everything else.